### ISSUE TITLE
🐛 Experience accordion header separator announced to screen readers

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#6610]
## 🐛 Bug

The vertical bar that separators the experience type and date is being announced to screen readers right now.

## 🦋 Expected Behaviour

Vertical bar is not announced to screen readers, or it is announced as a separator rather than "vertical bar".

## 🕵️ Details

The vertical bar provides no real value and detracts from the content.

## 📋 Steps to Reproduce

1. Fire up w/e screen reader you have on hand
2. Navigate to a page with an experience accordion
3. Tab focus over to the accordion trigger
4. Notice the vertical bar is announced to screen readers

## 🙋‍♀️ Proposed Solution

Any of these solutuions would work, however we first need to know if the separation has any value.

1. Swap out the `|` for `<Separator orientation="vertical" decorative />` Preferred if the separator has **no** real value
2. Swap out the `|` for `<Separator orientation="vertical" />` _Only_ if the separation has value other than visual separation
3. Add `aria-hidden` to the `span` that wraps the `|` 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/8ddf2929bc2348c3361f6109f7a46f2223863cd8/apps/web/src/components/ExperienceAccordion/ExperienceAccordionHeader.tsx#L53-L55

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] Vertical bar is still visually present
- [ ] Vertical bar is not announced to screen readers, or it is announced as a separator rather than "vertical bar"

**Comment by esizer** [comment]
Component removed in favour of `ExperienceCard`
